### ISSUE TITLE
Collection can't be opened with 2.12alpha16 / ConcurrentModificationException

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#6544]
###### Reproduction Steps

1. Have a collection with subdecks that have cards due etc
2. Open 2.12alpha16 on a real device but still hard to reproduce? Does not show up on an emulator
3. Can't open collection dialog

###### Expected Result

Works

###### Actual Result

```
06-23 20:22:58.772 18890 19562 E AnkiDroid: CollectionTask/ doInBackgroundLoadDeckCounts - error
06-23 20:22:58.772 18890 19562 E AnkiDroid: java.util.ConcurrentModificationException
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.ArrayList$Itr.next(ArrayList.java:860)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.Sched._groupChildrenMain(Sched.java:15)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.SchedV2._groupChildren(SchedV2.java:4)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:3)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:3)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:43)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.lang.Thread.run(Thread.java:919)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 
06-23 20:22:58.772 18890 19562 E AnkiDroid: java.util.ConcurrentModificationException
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.ArrayList$Itr.next(ArrayList.java:860)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.Sched._groupChildrenMain(Sched.java:15)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.SchedV2._groupChildren(SchedV2.java:4)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:3)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:3)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:43)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
06-23 20:22:58.772 18890 19562 E AnkiDroid: 	at java.lang.Thread.run(Thread.java:919)
06-23 20:22:58.785  1100  1419 I ActivityTaskManager: Displayed com.ichi2.anki.a/com.ichi2.anki.DeckPicker: +128ms

```

You can paste that stack trace in to Android Studio to make it clickable for the 'Analyze' top menu, 'Analyze stack trace or thread dump' at the bottom of the popup

The line numbers are bogus because R8 is minimizing them incorrectly and this was from a release build

@Arthur-Milchior seems like the groups of children can be modified at the same time they are iterated?

Related to #6539 I bet

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
Not sure what happened with #6539 but when I reverted it locally and tested, everything was fine on my device.

I will attach the collection that reproduced it here but I want to emphasize that it did not reproduce on the emulators, only on my device so I think it is a timing / async issue.

It shows the structure of the collection though, it's incredibly simple really - 8 decks total, but 4 of them are sub-decks under the same parent deck.

[20200623_Nico_2.12alpha16_crash.colpkg.zip](https://github.com/ankidroid/Anki-Android/files/4822788/20200623_Nico_2.12alpha16_crash.colpkg.zip)

**Comment by Arthur-Milchior** [comment]
It's absolutely not async/timing issue. It was a problem due to the fact that I was iterating on `children` and adding the elements on `children` in `children`, which made no sens. The real problem is that I cherry-edpick commits from my big PR and that this PR did depends on change made in other commits of the same PR. I corrected it in https://github.com/ankidroid/Anki-Android/pull/6547

**Comment by david-allison-1** [comment]
Fixed in #6547
